### PR TITLE
Add a PSR-4 loading script to allow Smarty to be used without Composer

### PR DESCRIPTION
**Pull request by scottchiefbaker**
This is to help address #999. I created a fake "class" script that acts as a PSR-4 loading script. For Smarty 3.x and 4.x I would instantiate Smarty with:

```PHP
require("include/smarty/libs/Smarty.class.php");
$smarty = new Smarty();
```

With this new script I can instantiate Smarty 5.x with:

```PHP
require("include/smarty/src/Smarty.class.php");
$smarty = new Smarty\Smarty();
```

This should help backwards compatible usability for old users like me. It'd be great if this script could live in `/libs/` to be 99% backwards compatible, but that directory doesn't exist anymore.

### CODE REVIEW

**Comment by scottchiefbaker** [comment]
Would you consider letting me recreate the `/libs/` directory and putting this file in there? That would make this *almost* a drop in replacement for Smart 3.x and 4.x for us non-composer users. Then the **only** difference would be:

```PHP
$smarty = new Smarty(); // Smarty 3.x and 4.x

// Versus

$smarty = new Smarty\Smarty(); // Smarty 5.x
```

**Comment by wisskid** [comment]
Yes, that seems fine!

**Comment by scottchiefbaker** [comment]
Excellent. I recreated the `libs/` folder and put the PSR-4 file in there. I've tested the changes and everything seems to be working now.

File: libs/Smarty.class.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	// Hack off the prefix part
+	$relative_class = substr($class, $len);
+
+	// Build a path to the include file
+	$file = __SMARTY_DIR . str_replace('\\', '/', $relative_class) . '.php';
+
+	// If the file exists, require it
+	if (file_exists($file)) {
+		//print "<div>Class <b>$class</b> maps to <code>$file</code></div>\n";
```
Comment (wisskid) [inline libs/Smarty.class.php:40]: Please remove debug code

File: libs/Smarty.class.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+		require_once($file);
+	}
+});
+
+// vim: tabstop=4 shiftwidth=4 noexpandtab autoindent softtabstop=4
```
Comment (wisskid) [inline libs/Smarty.class.php:46]: Please remove

**Review by wisskid — CHANGES_REQUESTED** [review]

**Comment by scottchiefbaker** [comment]
PR updated with your requested fixes.

**Comment by scottchiefbaker** [comment]
@wisskid do you need anything else from me on this?

**Review by wisskid — APPROVED** [review]

**Comment by wisskid** [comment]
@scottchiefbaker this has been merged (with a changelog) as https://github.com/smarty-php/smarty/pull/1019. Thanks!

**Pull request closed without merging**